utils/flink-components-testkit/src/main/scala/pl/touk/nussknacker/engine/flink/util/test/FlinkTestExceptionHolder.scala

```diff
@@ -0,0 +1,36 @@
+package pl.touk.nussknacker.engine.flink.util.test
+
+import pl.touk.nussknacker.engine.api.exception.NuExceptionInfo
+import pl.touk.nussknacker.engine.testmode.TestRunId
+
+final class FlinkTestExceptionHolder(testRunId: TestRunId) extends Serializable with AutoCloseable {
```

Comment:
Why do you need this if we already have `ResultsCollectingListener.exceptionThrown`. I thought that we only need to use already stored exceptions

Reply:
Oh, I didn't notice that class :) ResultsCollectingListener.exceptionThrown converts exception to `ExceptionResult[_]`. We can try to use it, but there will be an effort to reconvert these exceptions to NuExceptionInfo. I don't know which way is better. WDYT?

Reply:
During the converting process from NuExceptionInfo to ExceptionResult we lose some information like from NodeComponentInfo.componentInfo..